src/devices/GoPiGo3/GoPiGo3.cs

```diff
@@ -0,0 +1,900 @@
+        /// <param name="Value">The value to be sent</param>
+        public void SpiWrite16(SpiMessageType MessageType, int Value)
+        {
+            byte[] outArray = { SpiAddress, (byte)MessageType, (byte)((Value >> 8) & 0xFF), (byte)(Value & 0xFF) };
+            SpiTransferArray(outArray);
+        }
+
+        /// <summary>
+        /// Send a 24 bit value over SPI
+        /// </summary>
+        /// <param name="MessageType">The SPI message type</param>
+        /// <param name="Value">The value to be sent</param>
+        public void SpiWrite24(SpiMessageType MessageType, int Value)
+        {
+            byte[] outArray = { SpiAddress, (byte)MessageType, (byte)((Value >> 16) & 0xFF), (byte)((Value >> 8) & 0xFF), (byte)(Value & 0xFF) };
+            SpiTransferArray(outArray);
+        }
+
+        /// <summary>
+        /// Send a 32 bit value over SPI
+        /// </summary>
+        /// <param name="MessageType">The SPI message type</param>
+        /// <param name="Value">The value to be sent</param>
+        public void SpiWrite32(SpiMessageType MessageType, int Value)
+        {
+            byte[] outArray = { SpiAddress, (byte)MessageType, (byte)((Value >> 24) & 0xFF), (byte)((Value >> 16) & 0xFF), (byte)((Value >> 8) & 0xFF), (byte)(Value & 0xFF) };
+            SpiTransferArray(outArray);
+        }
+
+        #endregion
+
+        #region Board elements
+
+        /// <summary>
+        /// Read the 20 characters of GoPiGo3 manufacturer name
+        /// </summary>
+        /// <returns>Returns the GoPiGo3 manufacturer name string</returns>
+        public string GetManufacturer()
+        {
+            string retVal = string.Empty;
+            byte[] outArray = {SpiAddress, (byte)SpiMessageType.GetManufacturer,
+                  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+            byte[] reply = SpiTransferArray(outArray);
+            if (reply[3] != 0xA5)
+            {
+                throw new IOException("No SPI response");
+            }
+            else
+            {
+                for (int ndx = 4; ndx < 24; ++ndx)
+                {
+                    if (reply[ndx] != 0)
+                    {
+                        retVal += (char)reply[ndx];
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+            }
+            return retVal;
+        }
+
+        /// <summary>
+        /// Read the 20 characters of GoPiGo3 board name
+        /// </summary>
+        /// <returns>Returns the GoPiGo3 board name string</returns>
+        public string GetBoard()
+        {
+            string retVal = string.Empty;
+            byte[] outArray = {SpiAddress, (byte)SpiMessageType.GetName,
+                    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+            byte[] reply = SpiTransferArray(outArray);
+
+            if (reply[3] == 0xA5)
```

Comment:
should A5 be a constant?

Reply:
yes, I made it as a const now. It's the return in case data are corect